### PR TITLE
add version_bump.go script to automate the release pipeline

### LINKED ISSUE #7169: Bump versions using go script
This is a child issue to the parent issue https://github.com/cri-o/cri-o/issues/4003

### Description:
Currently, the version of ```cri-o``` needs to be manually modified in the ```internal/version/version.go``` file.

This needs to be automated using a go-script which can be used in the CI environments to bump that version automatically.

### Additional Context:
In the future, the modified version would need to be pushed to the main branch using GitHub action workflow automatically.

### PR DESCRIPTION
**Pull request by anshgoyalevil**
#### What type of PR is this?
This PR adds the ```version_bump.go``` script to automatically bump the version in internal/version/version.go file.
/kind feature

#### What this PR does / why we need it:
We needed this feature to automate the version bumping using GitHub actions. This script has little use as of now, but as per issue https://github.com/cri-o/cri-o/issues/4003 this script would be used in the CI environment of GitHub actions to change the version and create pull requests.

#### Which issue(s) this PR fixes:
fixes #7169 

#### Special notes for your reviewer:
This script is designed to work as below:

- run the command ```go run scripts/version_bump.go```
- The version inside the internal/version/version.go would be changed. For example: 1.2.3 to 1.2.4

It also supports additional arguments like:
- ```go run scripts/version_bump.go patch``` for patch bump. Example: 1.2.3 to 1.2.4
- ```go run scripts/version_bump.go minor``` for minor bump. Example: 1.2.3 to 1.3.0
- ```go run scripts/version_bump.go major``` for major bump. Example: 1.2.3 to 2.0.0

#### Does this PR introduce a user-facing change?
No

```release-note
add version bump automation script
```

### CODE REVIEW

File: scripts/version_bump.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"strconv"
+)
+
+const versionFile = "../internal/version/version.go"
```
Comment (haircommander) [inline scripts/version_bump.go:11]: nit: can this be an arg? It could make unit testing easier

File: scripts/version_bump.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+func main() {
+	var bumpType string
+	if len(os.Args) > 1 {
```
Comment (haircommander) [inline scripts/version_bump.go:15]: if we use the arg approach above, we may want this to be a flag or something to differentiate. Up to you though

**Comment by haircommander** [comment]
/ok-to-test

thanks @anshgoyalevil !

**Comment by anshgoyalevil** [comment]
Thanks for the inputs @haircommander . Would make changes by tomorrow, along with the fix to failing workflow. (Forgot to lint the code :smile:)

File: scripts/version_bump.go (hunk 1 above)
Comment (anshgoyalevil) [reply scripts/version_bump.go]: Done! Added an extra parameter ```-f``` which takes the path to file

File: scripts/version_bump.go (hunk 2 above)
Comment (anshgoyalevil) [reply scripts/version_bump.go]: Done! Added an extra parameter ```-bump``` which takes the three bump types ```minor```, ```major``` and ```patch```

**Comment by anshgoyalevil** [comment]
@haircommander I am sure the script is ready to merge. Though automated tests are not passing, as per the comments above.
Is there something from my side that need to be addressed for them to be passed?

**Review by saschagrunert — CHANGES_REQUESTED** [review]
The [linter fails](https://github.com/cri-o/cri-o/actions/runs/5664612854/job/15350933803?pr=7170):

```
Error: octalLiteral: use new octal literal style, 0o644 (gocritic)
  Error: paramTypeCombine: func(version string, bumpType string) string could be replaced with func(version, bumpType string) string (gocritic)
  Error: paramTypeCombine: func(versionFile string, newVersion string) error could be replaced with func(versionFile, newVersion string) error (gocritic)
  Error: sprintfQuotedString: use %q instead of "%s" for quoted strings (gocritic)
```

**Pull request closed without merging**